samples/posix/eventfd/prj.conf

```diff
@@ -1,5 +1,6 @@
 # General config
 CONFIG_NEWLIB_LIBC=y
+CONFIG_NEWLIB_LIBC_NANO=n
```

Comment:
Is `CONFIG_NEWLIB_LIBC_NANO=y` selected by default?

Reply:
As explained in https://github.com/zephyrproject-rtos/zephyr/pull/26187#issuecomment-644683198, different platforms select and deselect it randomly. (Well, not randomly, but based on the good intentions of platform maintainers. Such good intentions then lead to user confusion, in full accordance with a known proverb.)

Reply:
Nano is enabled by default when supported by toolchain. `CONFIG_HAS_NEWLIB_LIBC_NANO` controls this.

In terms of the Zephyr SDK, newlib nano libraries are available on all platforms, but `nano.specs` is not (https://github.com/zephyrproject-rtos/sdk-ng/issues/182). I am planning to address this soon.

Reply:
> I am planning to address this soon.

Good and thanks.

Because how it came about in this particular case is that my (as I hope everybody else's) default test platform is qemu_x86, where it works OOB, while, as was reported in #26187, on a typical ARM platform, it doesn't. It should be an OS policy to require, and enforce, consistent configuration for hardware-independent features. (And as we know there're multiple obstacles to that, as even hardware-independent things as libc are actually dependent on hardware bits and pieces.)